src/allmydata/util/nummedobj.py

```diff
@@ -1,7 +1,9 @@
-import collections, itertools
```

Comment:
All changes in this file motivated by a missing feature in the implementation of NummedObj which caused an exception to be raised if a NummedObj was ever compared to a non-NummedObj.  It's amazing this didn't happen previously but apparently that is the case.  Changes in this branch (specifically, a call to `disownServiceParent`) caused such a comparison to happen and this code to blow up.  Changes here are aimed at improving the general correctness of NummedObj.